src/history/file_backed.rs

```diff
@@ -59,7 +59,7 @@ impl History for FileBackedHistory {
             .entries
             .back()
             .map_or(true, |previous| previous != &entry)
-            && !entry.is_empty()
+            && !entry.is_empty() && self.capacity > 0
```

Comment:
If I understand this correctly, if capacity is 0, we never go into the block where we try to delete the oldest entry on save and we don't save this entry either. I guess that seems reasonable. Although, @sholderbach was talking about a sync issue so I'd like to get his input too.